### PR TITLE
[#415] Make sure no password is visible in the rendered configuration form

### LINKED ISSUE #415: Apigee Edge admin page shows password in plaintext within source
## Description
As a security concern, the Apigee Edge plugin config should not have the password in plain text within the source. 

## Steps to Reproduce
1. Login as an admin
2. Goto the Apigee Edge config
3. View source

## Actual Behavior
Password is shown in plain text within the source

## Expected Behavior
No password is ever shown. It may be saved but never retrieved. Or if its retrieved its done via a GET call.

## Version Info
Machine name: apigee_edge
Version: 8.x-1.6

### PR DESCRIPTION
**Pull request by minnur**

### CODE REVIEW

**Comment by cnovak** [comment]
I would rather follow the way SMTP module does the password where if you don't put it in it is left unchanged. This way if someone submits a change to the form they do not loose the current password. This could cause an outage on a production site.

File: src/Plugin/KeyInput/ApigeeAuthKeyInput.php [hunk 1]
```diff
@@ -259,6 +260,21 @@ public function validateConfigurationForm(array &$form, FormStateInterface $form
...
+      $key_input_plugin_form_state = clone $form_state;
+      $key_input_plugin_form_state->setValues($form_state->getUserInput()['key_input_settings']);
+      // @see \Drupal\key\Form\KeyFormBase::validateForm()
+      $key_input_processed_values = $form_state->getFormObject()->getEntity()->getKeyInput()->processSubmittedKeyValue($key_input_plugin_form_state);
+      $key_value = $key_input_processed_values['processed_submitted'];
+    }
+
+    // Could be an empty array.
+    $values = Json::decode($key_value);
+
```
Comment (arlina-espinoza) [inline src/Plugin/KeyInput/ApigeeAuthKeyInput.php]: Thanks @minnur , this is looking good. However something around this section is giving me a white screen when checking the scenario of the module just installed (no credentials have ever been set). Could you try to replicate?

**Review by arlina-espinoza — CHANGES_REQUESTED** [review]
See comment about the scenario of module just installed.

File: src/Plugin/KeyInput/ApigeeAuthKeyInput.php (hunk 1 above)
Comment (minnur) [reply src/Plugin/KeyInput/ApigeeAuthKeyInput.php]: @arlina-espinoza looks like the module is generating WSOD even without my changes :(

Comment (minnur) [reply src/Plugin/KeyInput/ApigeeAuthKeyInput.php]: @arlina-espinoza I think I got this working. Would you mind reviewing the changes?

**Comment by arlina-espinoza** [comment]
@googlebot I consent.

**Review by arlina-espinoza — APPROVED** [review]
Thanks @minnur, LGTM. I've also fixed the tests.

**Pull request merged**